### PR TITLE
chore: update web3auth deps and fix build

### LINKED ISSUE #2056: [Feature]: Update web3auth connector to latest version of their SDK
### Is your request related to a problem?

web3auth have deprecated some 'openlogin' networks and have added new ones that are not accessible via this release of blocknative.

This means client Ids that are using latest environments can not be connected via blocknative web3auth connector.

### Feature Description

Would be to release a new version of the web3auth package with latest versions, probably not too much work, I think mostly what changes is some constants are added/removed.

### Alternative Solutions

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by bengobeil**
### Description
resolve #2056 

### Checklist
  - [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

**Comment by bengobeil** [comment]
just have one question in the code pertaining to behavior for empty rpcUrl now that web3auth does not allow undefined

**Comment by bengobeil** [comment]
tested locally within our dapp
image

**Comment by bengobeil** [comment]
@Adamj1232 ready for review! (can't add a reviewer it seems)

File: packages/web3auth/src/index.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ function web3auth(options: Web3AuthModuleOptions): WalletInit {
           ticker: token,
           tickerName: label,
           chainId: id,
-          rpcTarget: rpcUrl,
+          rpcTarget: rpcUrl ?? "", // TODO: Ask maintainers if we should throw instead.
```
Comment (Adamj1232) [inline packages/web3auth/src/index.ts:51]: Is the rpcUrl prop required or are there defaults per chain for the web3Auth provider?

Comment (bengobeil) [reply packages/web3auth/src/index.ts:51]: I don't believe web3auth has defaults per chain.

Comment (bengobeil) [reply packages/web3auth/src/index.ts:51]: and `rpcTarget` is required.

Comment (bengobeil) [reply packages/web3auth/src/index.ts:51]: (at least according to the type system)

Comment (bengobeil) [reply packages/web3auth/src/index.ts:51]: OK their documentation is wrong, they do require an rpcTarget. I suggest passing the value to web3auth and let it fail on their end. It's going to throw no matter what.

image

**Comment by bengobeil** [comment]
@Adamj1232 ready for re-review, rpcUrl is required for web3auth

File: packages/web3auth/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/web3auth",
-  "version": "2.3.1",
+  "version": "2.4.0-alpha.1",
```
Comment (Adamj1232) [inline packages/web3auth/package.json:3]: Thanks @bengobeil  - We will also want to add this new version update to `packages/demo/package.json` and run `yarn && yarn check-all` from the root of the project and ensure there are not any errors

**Comment by bengobeil** [comment]
yarn check-all passes locally @Adamj1232 one check is failing but from what I can see it originates from another merged PR

**Review by Adamj1232 — APPROVED** [review]
Looks good, thanks for the PR!

**Comment by Adamj1232** [comment]
@bengobeil Im getting this error running the alpha branch - does `privateKeyProvider` need to be added to web3Auth init props?
Screenshot 2024-06-13 at 13 44 32

**Comment by bengobeil** [comment]
> @bengobeil Im getting this error running the alpha branch - does `privateKeyProvider` need to be added to web3Auth init props? Screenshot 2024-06-13 at 13 44 32

Yes, looks like this when I initialize it

image

**Comment by bengobeil** [comment]
> > @bengobeil Im getting this error running the alpha branch - does `privateKeyProvider` need to be added to web3Auth init props? Screenshot 2024-06-13 at 13 44 32
> 
> Yes, looks like this when I initialize it
> 
> image

from package @web3auth/ethereum-provider

**Comment by Adamj1232** [comment]
Ok well I will need to revert this merge as it doesnt look like it was tested with the app or updated for the necessary props

**Comment by bengobeil** [comment]
@Adamj1232 OK, I tried using the demo app with the current version (without my changes) and it didn't seem to work either. I think there is a react dependency.

I think the same props in a react app and it worked (the demo is with svelte).

As for the new props, how do you want to handle that?

**Comment by Adamj1232** [comment]
@bengobeil new props will need to be added to the `Web3AuthModuleInitOptions` type here - https://github.com/blocknative/web3-onboard/blob/4d47a5c8f1cb69ae335653ef63066823341a4463/packages/web3auth/src/index.ts#L11

Then after checking with the internal svelte demo that it works we can test again.
If there is a react dependency then it may need to be added to the web3auth package.json - possibly as a peer-dep depending on how its setup

**Comment by bengobeil** [comment]
@Adamj1232 it seems `privateKeyProvider` is already in the props as part of `Web3AuthOptions`
image

**Comment by bengobeil** [comment]
you should get a compiler error if it's not present

**Comment by bengobeil** [comment]
I will however try to get it working with the demo

**Pull request merged**